Update pretalx2wtd to retrieve speaker info and other cleanup.

mxsasha opened the pull request:
The script now does its own API calls, and has a bit more
logging. Twitter and website are added to the sessions file
for Portland 2020. Order has changed.

mxsasha committed:
Update pretalx2wtd to retrieve speaker info and other cleanup.

The script now does its own API calls, and has a bit more
logging. Twitter and website are added to the sessions file
for Portland 2020. Order has changed.

mxsasha commented:
Nobody seems to have an avatar now, so we can add that feature once some speakers have uploaded one. Shouldn't be too hard now that we're retrieving full speaker info already for each speaker.

plaindocs commented:
This looks awesome. 

At least one speaker does have an Avatar, Lucy Cary.
They are also commented out on the speaker page till we have more.

mxsasha commented:
Oh, actually it's already in there. Do we want to make our own download, or just defer to pretalx? I don't like to manage automatic downloads, so I'm leaning to just do it like this:

```
  speakers:
  - name: Lucy Carey
    slug: lucy-carey
    avatar: https://pretalx.com/media/avatars/Screenshot_2020-02-19_at_09.56.26_pUIrHqc.png
    twitter: lucyrushi
```

plaindocs commented:
I'd be tempted to run it manually and update the git repo with the images so we're not dependent on Pretalx at build time. 

So wget the image and save to > images/slug.img 

What do you think @mxsasha

mxsasha commented:
> I'd be tempted to run it manually and update the git repo with the images so we're not dependent on Pretalx at build time.

I'll add that.

mxsasha committed:
Add downloading of speaker avatars.

mxsasha commented:
I've added downloading of speaker images. Note that the location is derived from the slug in the speaker_photo template filter, so we don't actually need the avatar location in the sessions YAML, so I removed that. The speaker_photo template filter should be compatible with this.

mxsasha commented:
@plaindocs I think this is now ready for merging, can you approve the PR if you agree?

plaindocs commented on docs/_scripts/pretalx2wtd.py:
```diff
@@ -1,55 +1,110 @@
+        print('Error: PRETALX_TOKEN not found in environment variables.')
+        return
+    http_headers = {'Authorization': 'Token ' + os.environ['PRETALX_TOKEN']}
+    submissions_url = f'https://pretalx.com/api/events/{pretalx_slug}/submissions/?state=confirmed'
```
`{pretalx_slug}` needs to available here. Not sure if you use any of the other

mxsasha replied:
I am confused. It is available. The script runs fine?

mxsasha committed:
Update to latest Portland 2020 sessions

mxsasha commented:
@plaindocs I added the current output for the Portland 2020 sessions in this PR too. The script seems to run fine, and I think it was all done.

So I think we can merge this, which will also update the Portland 2020 sessions - you should probably have a look over https://github.com/writethedocs/www/blob/pretalx-improvements/docs/_data/portland-2020-sessions.yaml to see if it seems correct.

plaindocs committed:
Show pics

plaindocs reviewed (approved):
This is awesome!
